vizro-core/tests/unit/vizro/actions/test_set_control.py

```diff
@@ -163,14 +185,55 @@ def test_pre_build_target_model_on_different_page_show_in_url_false(self):
             ValueError,
             match=re.escape(
                 "Model with ID `filter_page_2_show_in_url_false` used as a `target` in `set_control` action is on a "
-                "different page than the trigger and must have `show_in_url=True`."
+                "different page from the trigger and so must have `show_in_url=True`."
             ),
         ):
             action.pre_build()
 
 
 @pytest.mark.usefixtures("managers_two_pages_for_set_control")
-@pytest.mark.filterwarnings("ignore:.*experimental.*:FutureWarning")
+class TestSetControlFunction:
+    """Tests set control function."""
+
+    def test_function_target_model_on_same_page(self):
+        # Add action to relevant component and set target to a control on the same page
+        action = set_control(target="filter_page_1", value="continent")
+        # Any other model that supports set_control can be used here, but the Graph used for the simplicity.
+        model_manager["scatter_chart_1"].actions = action
+        # Call pre_build to set _same_page attribute
+        action.pre_build()
+
+        # Call function method with a mock trigger value
+        result = action.function(_trigger={"points": [{"customdata": ["Europe"]}]})
+        expected = "Europe"
+
+        assert result == expected
+
+    def test_function_target_model_on_different_page(self, monkeypatch):
+        # Add action to relevant component and set target to a control on different page with show_in_url=True
+        action = set_control(target="filter_page_2_show_in_url_true", value="continent")
+        # Any other model that supports set_control can be used here, but the Graph used for the simplicity.
+        model_manager["scatter_chart_1"].actions = action
+        # Call pre_build to set _same_page attribute
+        action.pre_build()
+
+        # Mock dash.get_relative_path as it's used in set_control.function
+        monkeypatch.setattr(set_control_module, "get_relative_path", lambda path: "/mocked_path")
```

Comment:
Can we do this with `mocker` instead? That's normally how we do this. There's a `return_value` argument that we can use.